### PR TITLE
fix: check if processes is nil to avoid panic

### LINKED ISSUE #5873: CLI nil pointer on old dashboard
## Bug Report
Still had a dashboard command running while I deleted a server, and got the following nil pointer:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x555b822b6ad2]

goroutine 1 [running]:
github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/components.(*ProcessTable).Update(0xc000c45560, {0xc0009b8060?, 0xc000536000?}, 0xc000ebe000?)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/components/tables.go:74 +0xf2
github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard.(*UI).UpdateData(0xc0011f99a0)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/ui.go:202 +0x112
github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard.(*UI).Main(0xc0011f99a0, {0x555b8311a920, 0xc000d688a0}, 0xc0005d7200)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/ui.go:173 +0x3698
github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard.Main({0x555b8311a920, 0xc000d688a0}, 0xc000cfaf80, 0xb2d05e00)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/dashboard.go:27 +0x125
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.glob..func16.1({0x555b8311a920?, 0xc000d688a0?}, 0xc00017cff0?)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard.go:41 +0x2a
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.WithClient.func1({0x555b8311a878?, 0xc000d503c0?}, 0xc000cfaf80)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/root.go:98 +0x115
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.WithClientNoNodes.func1({0x555b8311a878, 0xc000d503c0})
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/root.go:74 +0x459
github.com/talos-systems/talos/pkg/cli.WithContext({0x555b8311a8b0?, 0xc000052170?}, 0xc000cc9cc0)
	github.com/talos-systems/talos/pkg/cli/context.go:40 +0x1d1
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.WithClientNoNodes(0xc000c3d600?)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/root.go:47 +0x45
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.WithClient(0x0?)
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/root.go:81 +0x35
github.com/talos-systems/talos/cmd/talosctl/cmd/talos.glob..func16(0x555b845bb800?, {0xc000d50380?, 0x4?, 0x4?})
	github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard.go:40 +0x25
github.com/spf13/cobra.(*Command).execute(0x555b845bb800, {0xc000d50340, 0x4, 0x4})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0x555b845b5e00)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/talos-systems/talos/cmd/talosctl/cmd.Execute()
	github.com/talos-systems/talos/cmd/talosctl/cmd/root.go:45 +0x1d5
main.main()
	github.com/talos-systems/talos/cmd/talosctl/main.go:18 +0x25
```

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

# Pull Request

## What? (description)
Check if `nodeData.Processes` is nil before trying to access `nodeData.Processes.Processes`

## Why? (reasoning)

This can lead to panic.

## Acceptance

Please use the following checklist:

- [x] you linked an issue (if applicable)
- [x] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)
 did not work locally, not sure why (not an issue with my branch, saw the same issue with master branch: `ERROR: failed to solve: granting entitlement security.insecure is not allowed by build daemon configuration`)

> See `make help` for a description of the available targets.

Fixes #5873

### CODE REVIEW

File: cmd/talosctl/cmd/talos/dashboard/components/tables_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"testing"
+	"time"
+
+	"github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/components"
+	"github.com/talos-systems/talos/cmd/talosctl/cmd/talos/dashboard/data"
+	"github.com/talos-systems/talos/pkg/machinery/api/common"
+	"github.com/talos-systems/talos/pkg/machinery/api/machine"
+)
+
+func TestUpdate(t *testing.T) {
```
Comment (ankitm123) [inline cmd/talosctl/cmd/talos/dashboard/components/tables_test.go:17]: Will convert to table driven tests, just want to be sure that it makes sense  ...

**Comment by smira** [comment]
> Not sure abt the `conform/commit/gpg-identity` check, will check the documentation.

thanks for your PR, this check make sure commit is signed by someone from the org, so we'll fix it up before merging.

File: api/resource/definitions/cluster/cluster.proto [hunk 2]
```diff
@@ -53,4 +53,3 @@ message MemberSpec {
   talos.resource.definitions.enums.MachineType machine_type = 4;
   string operating_system = 5;
 }
-
```
[the remark this answers is not in the document]
Comment (ankitm123) [reply api/resource/definitions/cluster/cluster.proto]: Done.

File: website/content/v1.3/reference/cli.md [hunk 3]
```diff
@@ -96,7 +96,7 @@ talosctl cluster create [flags]
       --bad-rtc                                  launch VM with bad RTC state (QEMU only)
```
[the remark this answers is not in the document]
Comment (ankitm123) [reply website/content/v1.3/reference/cli.md]: Ran
```
make USERNAME=siderolabs generate
```
and it reverted my changes. 
IMO, this should be added to the PR template, so new users dont run into this issue, wdyt?
PR template says to run `make generate` :thinking:

**Comment by ankitm123** [comment]
Are the unit tests flaky? They passed the first time :shrug:

**Review by smira — APPROVED** [review]
thank you!

**Pull request merged**